docs/gremlin_tutorials.md

```diff
@@ -0,0 +1,731 @@
+    ```bash
+    g.V().has("name") # find vertices containing property `name`
+    ```
+* propertyKey - the key of the property to filter on, </br> value - the value to compare the accessor value to for equality. 
+    ```bash
+    g.V().has("age", 10)
+    g.V().has("name", "marko")
+    g.E().has("weight", 1.0)    
+    ```
+* propertyKey - the key of the property to filter on, </br> predicate - the filter to apply to the key's value. 
+    ```bash
+    g.V().has("age", P.eq(10))
+    g.V().has("age", P.neq(10))
+    g.V().has("age", P.gt(10))
+    g.V().has("age", P.lt(10))
+    g.V().has("age", P.gte(10))
+    g.V().has("age", P.lte(10))
+    g.V().has("age", P.within([10, 20]))
+    g.V().has("age", P.without([10, 20]))
+    g.V().has("age", P.inside(10, 20))
+    g.V().has("age", P.outside(10, 20))
+    g.V().has("age", P.not(P.eq(10))) # = g.V().has("age", P.neq(10))
+    g.V().has("name", TextP.startingWith("mar"))
+    g.V().has("name", TextP.endingWith("rko"))
+    g.V().has("name", TextP.containing("ark"))
+    g.V().has("name", TextP.notStartingWith("mar"))
+    g.V().has("name", TextP.notEndingWith("rko"))
+    g.V().has("name", TextP.notContaining("ark"))
+    ```
+* label - the label of the Element, </br> propertyKey - the key of the property to filter on, </br> value - the value to compare the accessor value to for equality. 
+    ```bash
+    g.V().has("person", "id", 1) # = g.V().hasLabel("person").has("id", 1)
+    ```
+* label - the label of the Element, </br> propertyKey - the key of the property to filter on, </br> predicate - the filter to apply to the key's value. 
+    ```bash
+    g.V().has("person", "age", P.eq(10)) # = g.V().hasLabel("person").has("age", P.eq(10))
+    ```
+#### hasNot()
+The hasNot()-step is meant to filter graph elements based on the non-existence of properties.
+
+Parameters: </br>
+propertyKey - the key of the property to filter on for non-existence.
+```bash
+g.V().hasNot("age") # find vertices not-containing property `age`
+```
+#### [is()](https://tinkerpop.apache.org/docs/current/reference/#is-step)
+The is()-step is meant to filter the object if it is unequal to the provided value or fails the provided predicate.
+
+Parameters: </br>
+* value - the value that the object must equal.
+    ```bash
+    g.V().out().count().is(1)
+    ```
+* predicate - the filter to apply.
+    ```bash
+    g.V().out().count().is(P.eq(1))
+    ```
+#### [where(traversal)](https://tinkerpop.apache.org/docs/current/reference/#where-step)
+The where(traversal)-step is meant to filter the current object by applying it to the nested traversal.
+
+Parameters: </br>
+whereTraversal - the traversal to apply.
+```bash
+g.V().where(out().count())
+g.V().where(out().count().is(gt(0)))
+```
+#### where(predicate)
+The where(predicate)-step is meant to filter the traverser based on the predicate acting on different tags.
+
+Parameters: </br>
+* predicate - the predicate containing another tag to apply.
+    ```bash
```

Comment:
Please check the code in the preview link.
It is too long to display in a page. 
Maybe following our style guide to wrap at 80 chars is an option. 
Or please wrap it manually to make it graceful in a page :)